lib/sinatra/base.rb

```diff
@@ -239,6 +239,18 @@ class NotFound < NameError #:nodoc:
     def http_status; 404 end
   end
 
+  class IndifferentHash < Hash
+    def [](key)
+      value = super(key)
+      return super(key.to_s) if value.nil? && Symbol === key
```

Comment:
@b264 maybe I used `Symbol === key` because that is what is in the current sinatra codebase.